dxr/build.py

```diff
@@ -539,8 +538,26 @@ def index_file(tree, tree_indexers, path, es, index, jinja_env):
 
              'is_text': is_text,
 
+             # if the file's an image, use <img src=image_src>
+             'image_src': image_src,
+
              'sections': build_sections(links)})
 
+def index_image(tree, rel_path, index, es):
+    '''
+    Index the image located at path relative to given tree using es in index.
+    Return the URL to get the image.
+    '''
+    # define the image source
+    image_src = "%s/images/%s" % (tree.name, rel_path)
+    # encode the image file as base64 string and index it
+    with open(join(tree.source_folder, rel_path), "rb") as img:
+        imgdata = img.read().encode("base64")
+        es.index(index, IMAGE, {
```

Comment:
Best of all, we wouldn't need this special-purpose stuff in build.py anymore; the image data would become merely a file-wide needle, returned from the core plugin's needles() method!